api/proto/solaris/v1/solaris.proto

```diff
@@ -55,11 +55,18 @@ message AppendRecordsRequest {
   string logID = 1;
   // records the list of records to be added
   repeated Record records = 2;
+  // returnIDs if true - response will contain list of inserted message IDs
+  bool returnIDs = 3;
 }
 
 // AppendRecordsResult contains the number or records added to the log
 message AppendRecordsResult {
+  // added is number of records added to database. If this value is less than passed ID list
   int64 added = 1;
+  // firstID is ID of first added record
+  string firstID = 2;
```

Comment:
If you didn't request id expanding, you will have only number of wrote messages. It's not enough to match records in database (if you want to do it somehow). But id of first message + number of records allows you to fetch exact messages.

Reply:
I have another "crazy" idea which is overkill, but I need to say - instead of flag of whether return the list of IDs or not, we can provide - how many new IDs to return :).

Reply:
It can be a good approach, but we need proper behavior for "return all"